src/plugins/visualizations/public/saved_visualizations/saved_visualization_references/timeseries_references.ts

```diff
@@ -13,7 +13,7 @@ import { VisParams } from '../../../common';
 const REF_NAME_POSTFIX = '_ref_name';
 
 /** @internal **/
-const INDEX_PATTERN_REF_TYPE = 'index_pattern';
+const INDEX_PATTERN_REF_TYPE = 'index-pattern';
```

Comment:
I haven't tested it and gone through the code. I just have an idea. Maybe we could export this ref type from the data-plugin and also use this here https://github.com/elastic/kibana/blob/master/src/plugins/data/common/search/search_source/extract_references.ts

Reply:
ok, done